Source/Diffterm.H

```diff
@@ -11,15 +11,24 @@
 #include "GradUtil.H"
 #include "Diffusion.H"
 
-// This header file contains functions and declarations for diffterm in 3D for
-// PeleC GPU. As per the convention of AMReX, inlined device functions are
-// defined here. Where as non-inline host functions are declared here.
+// This header file contains functions and declarations for diffterm.
 
-// The original Diffterm function passes too many arguments, which contribute to
-// kernel launch overhead. In order to combat this, we split some of the
-// computation up into separate Device functions which will then be launched by
-// a CPU function called pc_diffterm. pc_diffusion_flux calculates the diffusion
-// flux per diction.
+namespace {
+#ifdef AMREX_USE_GPU
+AMREX_GPU_CONSTANT const amrex::Real d_weights[3] = {0.0, 1.0, 0.5};
+#endif
+const amrex::Real h_weights[3] = {0.0, 1.0, 0.5};
+
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE amrex::Real
+weight(const int n)
+{
+#if AMREX_DEVICE_COMPILE
+  return d_weights[n];
```

Comment:
Can it just return this with no ifdef?

Reply:
yes of course. I was being dumb and making this HOST_DEVICE. This is just a device function...